control/control.go

```diff
@@ -73,19 +72,14 @@ func (srv HTTPServer) Validate() error {
 }
 
 // Run executes the event loop for the control server
-func (srv *HTTPServer) Run(bus *events.EventBus) {
-	srv.Subscribe(bus, true)
-	srv.Bus = bus
-	srv.Start()
+func (srv *HTTPServer) Run(ctx context.Context, bus *events.EventBus) {
+	srv.Start(bus)
 
 	go func() {
-		defer srv.Stop()
+		defer srv.Stop(ctx)
 		for {
-			event := <-srv.Rx
-			switch event {
-			case
-				events.QuitByClose,
-				events.GlobalShutdown:
```

Comment:
Because we don't need it we can actually replace the whole goroutine with:

```
go func() {
	 defer srv.Stop(ctx)
	 <- ctx.Done()
	 return
}()
```